pkg/backup/compaction_job.go

```diff
@@ -523,17 +523,46 @@ func updateCompactionBackupDetails(
 	return compactedDetails, nil
 }
 
-// compactIntroducedSpans takes a compacted backup manifest and the full chain of backups it belongs
-// to and computes the introduced spans for the compacted backup.
-func compactIntroducedSpans(
-	ctx context.Context, manifest backuppb.BackupManifest, chain compactionChain,
-) (roachpb.Spans, error) {
-	if err := checkCoverage(ctx, manifest.Spans, chain.backupChain); err != nil {
+// createCompactionManifest creates a new manifest for a compaction job and its
+// compacted chain. The details should have its targets resolved.
+func (c compactionChain) createCompactionManifest(
+	ctx context.Context, details jobspb.BackupDetails,
+) (*backuppb.BackupManifest, error) {
+	// TODO (kev-cao): Will need to update the SSTSinkKeyWriter to support
+	// range keys.
+	lastBackup := c.lastBackup()
+	if len(lastBackup.Tenants) != 0 {
+		return nil, errors.New("backup compactions does not yet support range keys")
+	}
+	introducedSpans, err := c.compactIntroducedSpans(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cManifest := lastBackup
+	cManifest.StartTime = c.start
+	cManifest.EndTime = c.end
+	cManifest.Descriptors = details.ResolvedTargets
+	cManifest.CompleteDbs = details.ResolvedCompleteDbs
+	cManifest.IntroducedSpans = introducedSpans
+	cManifest.Files = nil
+	cManifest.DescriptorChanges = nil // Revision history is not supported.
+	cManifest.EntryCounts = roachpb.RowCount{}
+	cManifest.Dir = cloudpb.ExternalStorage{}
+	cManifest.HasExternalManifestSSTs = false
```

Comment:
why is this false?

Reply:
Ran into an issue in `getCompletedSpans`/`genSpans` when they call `NewFileIter` and it can't find those external manifest SSTs since they haven't been written yet.

Reply:
could you add a docstring that explains that this is set to false as it will be set to true when the manifest is written to cloud storage?